src/java/org/apache/cassandra/tools/CompactionLogAnalyzer.java

```diff
@@ -0,0 +1,573 @@
+        int bucketsAboveT2;
+
+        /**
+         * Called to aggregate data in response to a new data point for a bucket.
+         * Unless the process is just starting, the new data point will be replacing the older state of the bucket,
+         * thus this will add the new data but also remove the older values.
+         */
+        private void updateTotals(DataPoint toAdd, DataPoint toRemove)
+        {
+            timestamp = toAdd.timestamp;
+            compactionsInProgress += toAdd.compactionsInProgress - toRemove.compactionsInProgress;
+            compactionsPending += toAdd.compactionsPending - toRemove.compactionsPending;
+            sstables += toAdd.sstables - toRemove.sstables;
+            size += toAdd.size - toRemove.size;
+            readBytesPerSecond += toAdd.readBytesPerSecond - toRemove.readBytesPerSecond;
+            writeBytesPerSecond += toAdd.writeBytesPerSecond - toRemove.writeBytesPerSecond;
+            totalBytes += toAdd.totalBytes - toRemove.totalBytes;
+            remainingReadBytes += toAdd.remainingReadBytes - toRemove.remainingReadBytes;
+            bucketsAboveT += toAdd.bucketsAboveT - toRemove.bucketsAboveT;
+            bucketsAboveT2 += toAdd.bucketsAboveT2 - toRemove.bucketsAboveT2;
+        }
+    }
+
+
+    final static Pattern CSVNamePattern = Pattern.compile("compaction-(\\w+)-(.*?)-(.*?)-(.*)\\.csv");
+    final static Pattern HumanReadablePattern = Pattern.compile("(\\d+(\\.\\d+)?)([ KMGTP])iB(/s)?");
+    final static String HumanReadablePowers = " KMGTP";
+    private static final String dateFormatter = "HH:mm:ss.SSS";
+    private static final String fullDateFormatter = "yyyy-MM-dd' 'HH:mm:ss.SSS";
+
+    static int reportResolutionInMs;
+
+    // Indexes of the relevant columns in the source CSV, set by initializeIndexes below.
+    static int timestampIndex = -1;
+    static int eventIndex;
+    static int bucketIndex;
+    static int sstablesIndex;
+    static int compactingSstablesIndex;
+    static int sizeIndex;
+    static int compactionsIndex;
+    static int readPerSecIndex;
+    static int writePerSecIndex;
+    static int sizesIndex;
+    static int Tindex;
+
+    private static void initializeIndexes(String header)
+    {
+        if (timestampIndex < 0)
+            synchronized (CompactionLogAnalyzer.class) {
+                if (timestampIndex < 0)
+                {
+                    Map<String, Integer> indexMap = new HashMap<>();
+                    String[] headers = header.split(",");
+                    for (int i = 0; i < headers.length; ++i)
+                        indexMap.put(headers[i], i);
+
+                    timestampIndex = indexMap.get("Timestamp");
+                    eventIndex = indexMap.get("Event");
+                    bucketIndex = indexMap.get("Bucket");
+                    // Note: this reads the current CSV format as well as an older one, where some of the headers of the
```

Comment:
We should get rid of the older header parsing here.

Reply:
Fixed in f7c74f02e7a049b72d1f150cca5dd46f437d1b30.